File: generate-print-design.js
const { createCanvas } = require('canvas');
const fs = require('fs');

// Create full resolution canvas (3600x2400 = 12" x 8" at 300 DPI)
const canvas = createCanvas(3600, 2400);
const ctx = canvas.getContext('2d');

// Design specifications matching your mockup
const width = 3600;
const height = 2400;
const baseFontSize = 120;

// Clear canvas (transparent background)
ctx.clearRect(0, 0, width, height);

// Set up text properties
ctx.fillStyle = '#1a1a1a';
ctx.textBaseline = 'middle';

const fontSize = baseFontSize;
const lineHeight = fontSize * 1.65;
const labelWidth = fontSize * 5.8;

let y = fontSize * 1.5;

// Top rule
ctx.strokeStyle = '#1a1a1a';
ctx.lineWidth = 2;
ctx.beginPath();
ctx.moveTo(0, y);
ctx.lineTo(width, y);
ctx.stroke();

y += fontSize * 1.8;

// From:
ctx.font = `700 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('From:', 0, y);
ctx.font = `400 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('Elon Musk <', labelWidth, y);

// Redacted email
const redactX = labelWidth + ctx.measureText('Elon Musk <').width;
const redactWidth = fontSize * 7.9;
ctx.fillRect(redactX, y - fontSize * 0.45, redactWidth, fontSize * 0.75);
ctx.fillText('>', redactX + redactWidth + fontSize * 0.15, y);

y += lineHeight;

// Sent:
ctx.font = `700 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('Sent:', 0, y);
ctx.font = `400 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('Sunday, November 25, 2012 12:36 AM', labelWidth, y);

y += lineHeight;

// To:
ctx.font = `700 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('To:', 0, y);
ctx.font = `400 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('Jeffrey Epstein', labelWidth, y);

y += lineHeight;

// Subject:
ctx.font = `700 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('Subject:', 0, y);
ctx.font = `400 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('RE:', labelWidth, y);

y += lineHeight * 2.2;

// Body
ctx.font = `400 ${fontSize}px Calibri, 'Segoe UI', sans-serif`;
ctx.fillText('What day/night will be the wildest party on your island?', 0, y);

// Save the canvas as PNG
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('print-design.png', buffer);

console.log('✅ Print-ready design created: print-design.png');
console.log('   Dimensions: 3600 x 2400 pixels (12" x 8" at 300 DPI)');
console.log('   Ready to upload to Gelato!');

File: generate-print-design.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFont
import sys

# Create canvas at 300 DPI (3600x2400 = 12" x 8")
width, height = 3600, 2400
img = Image.new('RGBA', (width, height), (255, 255, 255, 0))
draw = ImageDraw.Draw(img)

# Colors
text_color = (26, 26, 26, 255)

# Font settings
base_font_size = 120
line_height = int(base_font_size * 1.65)
label_width = int(base_font_size * 5.8)

# Try to load system fonts
try:
    font_bold = ImageFont.truetype("/System/Library/Fonts/Supplemental/Calibri Bold.ttf", base_font_size)
    font_regular = ImageFont.truetype("/System/Library/Fonts/Supplemental/Calibri.ttf", base_font_size)
except:
    try:
        font_bold = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", base_font_size)
        font_regular = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", base_font_size)
    except:
        # Fallback to default
        font_bold = ImageFont.load_default()
        font_regular = ImageFont.load_default()
        print("⚠️  Warning: Could not load Calibri font, using fallback")

y = int(base_font_size * 1.5)

# Top rule
draw.line([(0, y), (width, y)], fill=text_color, width=2)
y += int(base_font_size * 1.8)

# From:
draw.text((0, y), 'From:', fill=text_color, font=font_bold, anchor='lm')
draw.text((label_width, y), 'Elon Musk <', fill=text_color, font=font_regular, anchor='lm')

# Redacted email
redact_x = label_width + draw.textlength('Elon Musk <', font=font_regular)
redact_width = base_font_size * 7.9
redact_y_top = y - base_font_size * 0.45
redact_y_bottom = y + base_font_size * 0.3
draw.rectangle(
    [(redact_x, redact_y_top), (redact_x + redact_width, redact_y_bottom)],
    fill=text_color
)
draw.text((redact_x + redact_width + base_font_size * 0.15, y), '>', fill=text_color, font=font_regular, anchor='lm')

y += line_height

# Sent:
draw.text((0, y), 'Sent:', fill=text_color, font=font_bold, anchor='lm')
draw.text((label_width, y), 'Sunday, November 25, 2012 12:36 AM', fill=text_color, font=font_regular, anchor='lm')
y += line_height

# To:
draw.text((0, y), 'To:', fill=text_color, font=font_bold, anchor='lm')
draw.text((label_width, y), 'Jeffrey Epstein', fill=text_color, font=font_regular, anchor='lm')
y += line_height

# Subject:
draw.text((0, y), 'Subject:', fill=text_color, font=font_bold, anchor='lm')
draw.text((label_width, y), 'RE:', fill=text_color, font=font_regular, anchor='lm')
y += int(line_height * 2.2)

# Body
draw.text((0, y), 'What day/night will be the wildest party on your island?', fill=text_color, font=font_regular, anchor='lm')

# Save as PNG
img.save('print-design.png', 'PNG', dpi=(300, 300))
print('✅ Print-ready design created: print-design.png')
print('   Dimensions: 3600 x 2400 pixels (12" x 8" at 300 DPI)')
print('   Ready to upload to Gelato or use in automation!')
